### DISCUSSION TITLE
source code read problem.

### DISCUSSION THREAD

**Discussion by pipipapi** [moby/moby#46277] [Q&A]
pulll code from gitee mirror of github. 
execute make command okay . this no trouble . 
<img width="1400" alt="image" src="https://github.com/moby/moby/assets/29531009/48260422-8518-46a1-8475-68055b05b3ee">

but so many golang files seems have faults .  red. it's not link correct source file.
why?

**Comment by corhere** [comment]
Because you're using macOS. We have several source files with `//go:build linux` or similar build constraints, without any alternative for darwin. As far as your IDE knows, the symbols defined in those files don't exist. Unfortunately, getting your IDE to cooperate won't be as simple as configuring it to analyze the code with `GOOS=linux` as several symbols in certain packages come from cgo, which extracts their definitions from C header files. I personally develop on a macOS system and have to date been unsuccessful in setting up my IDE to analyze the sources without reporting spurious errors.
